cookbook-front-end/src/App.tsx

```diff
@@ -127,17 +110,30 @@ const App = () => {
           }
         </Toolbar>
       </AppBar>
-      {data.currentPanel === Panels.browseRecipes && <Recipes/>}
-      {data.currentPanel === Panels.recipeDetail && <RecipeDetail recipeDetailId={data.recipeDetailId}/>}
-      {data.currentPanel === Panels.createRecipe && data.userIsLoggedIn && <CreateRecipe/>}
+
+      <Switch>
+        <Route exact path="/recipes">
+          <Recipes />
+        </Route>
+        <Route exact path="/">
+          <Recipes />
+        </Route>
+        <Route path="/recipes/:recipeDetailId">
```

Comment:
This `:recipeDetailId` url parameter is cool. You can see in `RecipeDetail` how we can access it in the component.